File: packages/java-with-retrofit/FormsparkService.java
private class ContactModel {
    private String name;
    private String message;

    public String getName() {
        return name;
    }

    public String getMessage() {
        return message;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}


private interface FormsparkApi {

    @Headers({"Content-Type: application/json", "Accept: application/json"})
    @POST("your-form-id")
    Call<BasicResponseModel> contact(@Body ContactModel contactModel);

}

public class FormsparkService {

    private void submit() {
        Gson gson = new GsonBuilder()
                .setLenient()
                .create();

        OkHttpClient okHttpClient = new OkHttpClient().newBuilder()
                .connectTimeout(1000, TimeUnit.SECONDS)
                .readTimeout(1000, TimeUnit.SECONDS)
                .writeTimeout(1000, TimeUnit.SECONDS)
                .build();


        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://submit-form.com/")
                .client(okHttpClient)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build();


        RetrofitApi retrofitApi = retrofit.create(FormsparkApi.class);

        ContactModel contactModel = new ContactModel();
        contactModel.setName("your-name");
        contactModel.setMessage("your-message");

        call = retrofitApi.contactUs(contactModel);

        call.enqueue(new Callback<BasicResponseModel>() {
            @Override
            public void onResponse(Call<BasicResponseModel> call, Response<BasicResponseModel> response) {
                progressBar.setVisibility(View.GONE);
                runningService = false;
                if (response.isSuccessful()) {
                    // Request succesful

                } else {
                    // request not succesful
                }

            }

            @Override
            public void onFailure(Call<BasicResponseModel> call, Throwable t) {
                if (call.isCanceled()) {
                    // Request canceled
                } else {
                    // Request failed
                }
            }
        });
    }

}

File: packages/kotlin-with-retrofit/FormsparkService.kt
data class ContactResponse(val name: String, val message: String)

interface ApiService {

    @POST("your-form-id")
    fun contact(@Body data: HashMap<String, Any>): Single<ContactResponse>

}

class FormsparkService {

    companion object {
        private var instance: ServiceBuilder? = null
        private var retrofit: ApiService? = null
        private var context: Context? = null

        private const val baseURL = "https://submit-form.com/"

        fun initialize(context: Context) {
            if (instance == null) {
                instance = ServiceBuilder()
                Companion.context = context
                instance!!.initializeRetro()
            }
        }

        fun get(): ApiService {
            return retrofit
                    ?: throw IllegalStateException("FormsparkService must be initialized first.")
        }
    }

    private fun initializeRetro() {
        retrofit = Retrofit.Builder()
                .baseUrl(baseURL)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create(GsonBuilder().setLenient().create()))
                .client(getOkHttpClient())
                .build()
                .create(ApiService::class.java)
    }

    private fun getOkHttpClient(): OkHttpClient {
        return OkHttpClient
                .Builder()
                .connectTimeout(1000, TimeUnit.SECONDS)
                .writeTimeout(1000, TimeUnit.SECONDS)
                .readTimeout(1000, TimeUnit.SECONDS)
                .addInterceptor { chain ->
                    val requestBuilder: Request.Builder = chain.request().newBuilder()
                    requestBuilder.header("Content-Type", "application/json")
                    requestBuilder.header("Accept", "application/json")
                    chain.proceed(requestBuilder.build())
                }.build()
    }

}

// Then call: FormsparkService.get().contact(data)